multimap/src/main/java/org/infinispan/multimap/impl/HashMapBucket.java

```diff
@@ -80,13 +82,14 @@ public boolean isEmpty() {
       return values.isEmpty();
    }
 
-   public int removeAll(Collection<K> keys) {
+   public HashMapBucketResponse<Integer> removeAll(Collection<K> keys) {
       int res = 0;
+      Map<MultimapObjectWrapper<K>, V> copied = new HashMap<>(values);
```

Comment:
This should be more efficient to use an iterator and copy the entries into the copy instead of copy and then removing. Also then we won't have to create a MultimapObjectWrapper per key as we can directly compare the keys.

Reply:
I can create a static equals method on the `MultimapObjectWrapper` class for that, though.